support-figma/extended-layout-plugin/code.ts

```diff
@@ -707,12 +718,93 @@ else if (figma.command === "move-plugin-data") {
     ]
   }
 
+  function deltaTransformPoint(transform: Transform, point: Vector)  {
+    var x = point.x * transform[0][0] + point.y * transform[0][1];
+    var y = point.x * transform[1][0] + point.y * transform[1][1];
+    return { x, y };
+  }
+
+  // Hypotenuse of right triangle with sides x and y
+  function hypot(x: number, y: number): number {
+    return Math.sqrt(x * x + y * y)
+  }
+
+  function decomposeTransform(t: Transform) {
+    let result: any = {};
+    let row0x = t[0][0];
+    let row0y = t[0][1];
+    let row1x = t[1][0];
+    let row1y = t[1][1];
+    result.translateX = t[0][2];
+    result.translateY = t[1][2];
+
+    // Compute scaling factors.
+    result.scaleX = hypot(row0x, row0y)
+    result.scaleY = hypot(row1x, row1y)
+
+    // If determinant is negative, one axis was flipped.
+    const determinant = row0x * row1y - row0y * row1x
+    if (determinant < 0) {
+        // Flip axis with minimum unit vector dot product.
+        if (row0x < row1y) result.scaleX = -result.scaleX 
+        else result.scaleY = -result.scaleY
+    }
+
+    // Renormalize matrix to remove scale.
+    if (result.scaleX != 1) {
+        row0x *= 1 / result.scaleX
+        row0y *= 1 / result.scaleX
+    }
+    if (result.scaleY != 1) {
+        row1x *= 1 / result.scaleY
+        row1y *= 1 / result.scaleY
+    }
+
+    // Compute rotation and renormalize matrix.
+    result.angle = Math.atan2(row0y, row0x)
+
+    if (result.angle != 0) {
+        // Rotate(-angle) = [cos(angle), sin(angle), -sin(angle), cos(angle)]
+        //                = [row0x, -row0y, row0y, row0x]
+        // Thanks to the normalization above.
+        const sn = -row0y
+        const cs = row0x
+        const m11 = row0x
+        const m12 = row0y
+        const m21 = row1x
+        const m22 = row1y
+
+        row0x = cs * m11 + sn * m21
+        row0y = cs * m12 + sn * m22
+        row1x = -sn * m11 + cs * m21
+        row1y = -sn * m12 + cs * m22
+    }
+
+    result.m11 = row0x
+    result.m12 = row0y
+    result.m21 = row1x
+    result.m22 = row1y
+
+    // Convert into degrees because our rotation functions expect it.
+    result.angle = radiansToDegrees(result.angle);
+
+		// calculate delta transform point
+		const px = deltaTransformPoint(t, { x: 0, y: 1 });
```

Comment:
funny indent here -- wonder if we can get eslint going on the plugin sources at some point...

Reply:
Oops those got inserted as tabs, probably from copying some code. I'll fix in another commit as I have another change incoming soon.